test/cql-pytest/test_using_timeout.py

```diff
@@ -152,3 +153,15 @@ def invalid(stmt):
     invalid(f"SELECT * FROM {table} USING TIMEOUT 60s AND TTL 10000")
     invalid(f"SELECT * FROM {table} USING TIMEOUT 60s AND TTL 123 AND TIMESTAMP 911")
     invalid (f"DELETE FROM {table} USING TIMEOUT 60s AND TTL 42 WHERE p = 42")
+
+def test_truncate_using_timeout(scylla_only, cql, table1):
+    table = table1
+    key = unique_key_int()
+    cql.execute(f"INSERT INTO {table} (p,c,v) VALUES ({key},1,1) USING TIMEOUT 60m AND TTL 1000000 AND TIMESTAMP 321")
+    res = list(cql.execute(f"SELECT ttl(v), writetime(v) FROM {table} WHERE p = {key} and c = 1"))
+    assert len(res) == 1 and res[0].ttl_v > 0 and res[0].writetime_v == 321
+    cql.execute(f"TRUNCATE TABLE {table} USING TIMEOUT 1000s")
+    res = list(cql.execute(f"SELECT ttl(v), writetime(v) FROM {table} WHERE p = {key} and c = 1"))
+    assert len(res) == 0
+    with pytest.raises(NoHostAvailable):
+        cql.execute(f"TRUNCATE TABLE {table} USING TIMEOUT 0s")
```

Comment:
That's a good point.  In a sense, this is similar to writing with CL=ALL, so `NoHostAvailable` does makes sense to me.

However, in https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-debug/33/testReport/materialized_views_test/TestMaterializedViews/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split026___test_truncate_base_during_mv_insert_test_with_auto_snapshot/
I see that the dtest got this error:
```
cassandra.OperationTimedOut: errors={<Host: 127.0.52.3:9042 dc1>: <Error from server: code=1003 [Error during truncate] message="Error during truncate: seastar::rpc::timeout_error (rpc call timed out)">, <Host: 127.0.52.2:9042 dc1>: <Error from server: code=1003 [Error during truncate] message="Error during truncate: seastar::rpc::timeout_error (rpc call timed out)">}, last_host=127.0.52.1:9042
```

so I'll see if NoHostAvailable stemmed from setting the timeout to 0 in particular.
(since I didn't change anything on the timeout handling in the truncate path from the storage_proxy down)

Reply:
Does a write with CL=ALL which times out not return a WriteTimeout? If not, what is WriteTimeout (I just saw it's available in the driver's list of errors)? :-(

Reply:
I think the answer is: "it's complicated".
For example, see https://github.com/scylladb/scylla-dtest/blob/3bc3ea944c1c19ee12d707408ce757977948413a/tools/assertions.py#L74-L85

Reply:
The above example is just to show the proliferation of errors in cassandra.
(not directly related to the timeout case).

In any case, I think that the error type is orthogonal to this change and I suggest that we deal with it later, under a different GH issue.

Reply:
Ok.